contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,910 @@
+            _removeLiquidity(shareOfWethToRemove);
+        }
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw WETH and sends it to the Vault.
+     */
+    function withdrawAll() external override onlyVault nonReentrant {
+        if (tokenId > 0) {
+            _removeLiquidity(1e18);
+        }
+
+        uint256 _balance = IERC20(WETH).balanceOf(address(this));
+        if (_balance > 0) {
+            _withdraw(vaultAddress, _balance);
+        }
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+    }
+
+    function _withdraw(address _recipient, uint256 _amount) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Collect the AERO token from the gauge
+     */
+    function _collectRewardTokens() internal override {
+        if (tokenId > 0) {
+            clGauge.getReward(tokenId);
+        }
+        super._collectRewardTokens();
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     */
+    function _abstractSetPToken(address, address) internal override {
+        // the deployer shall call safeApproveAllTokens() to set necessary approvals
+        revert("Unsupported method");
+    }
+
+    /**
+     * @dev Approve the spending of all assets
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        // to add liquidity to the clPool
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        // to be able to rebalance using the swapRouter
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
```

Comment:
Our vault will check each asset the vault supports against each strategy. What we do in other strategies is `return 0 ` on unsupported assets. Otherwise if we added a second asset to the vault, this require would cause the vault to then revert mints/redeems/rebases.

Reply:
I think the vault only checks for assets a specific strategy supports: 
https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/vault/VaultCore.sol#L470-L472

Here is an example of Convex strategy that would revert on an unsupported asset: 
https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/ConvexStrategy.sol#L131

Reply:
Okay. We can leave it. No idea how I missed that.